src/prefect/settings.py

```diff
@@ -482,6 +482,23 @@ def default_cloud_ui_url(settings, value):
 # Setting definitions
 
 
+PREFECT_LIMIT_EXCEPTION_TO_STATE_MESSAGE = Setting(
```

Comment:
Since these settings are mutually exclusive I'm inclined to think they should be one setting. 
Maybe something like:
```
PREFECT_STATE_MESSAGE_EXCEPTION = Setting(
    Literal["full", "type_only", "hidden"],
    default="full"
)
```